### PR TITLE
fix: fiscal year should always be for 12 months

### PR DESCRIPTION
**Pull request by rohitwaghchaure**

### CODE REVIEW

File: erpnext/accounts/doctype/fiscal_year/fiscal_year.py [hunk 1]
```diff
@@ -37,9 +37,8 @@ def validate_dates(self):
 		if getdate(self.year_start_date) > getdate(self.year_end_date):
 			frappe.throw(_("Fiscal Year Start Date should not be greater than Fiscal Year End Date"))
 
-		if (getdate(self.year_end_date) - getdate(self.year_start_date)).days > 366:
-			date = getdate(self.year_start_date) + relativedelta(years=1) - relativedelta(days=1)
-			self.year_end_date = date.strftime("%Y-%m-%d")
+		date = getdate(self.year_start_date) + relativedelta(years=1) - relativedelta(days=1)
```
[the remark this answers is not in the document]
Comment (nabinhait) [reply erpnext/accounts/doctype/fiscal_year/fiscal_year.py]: @rohitwaghchaure Yes, don't change year end date silently. Give a proper feedback to the user, the user might set wrong start date as well.

**Review by nabinhait — DISMISSED** [review]

File: erpnext/accounts/doctype/fiscal_year/fiscal_year.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rohitwaghchaure) [reply erpnext/accounts/doctype/fiscal_year/fiscal_year.py]: Done!

**Comment by tonto0** [comment]
Needs to be reversed see
https://discuss.erpnext.com/t/fiscal-year-transition-to-calendar-year-required-by-law/56541

**Comment by martindehh** [comment]
I do not get wy this 'Feature' was built in anyways. It is possible and sometimes desired to have the ability to have longer or shorter fiscal years than 12 months.
Please revert this.

**Pull request merged**